pkg/dashboard/resource/project.go

```diff
@@ -39,6 +43,30 @@ type projectInfo struct {
 	Spec *platform.ProjectSpec `json:"spec,omitempty"`
 }
 
+type functionImportMap map[string]*functionInfo
+
+func (fim *functionImportMap) UnmarshalJSON(data []byte) error {
+	var functionImports map[string]*struct {
+		Function *functionInfo
+	}
+	if err := json.Unmarshal(data, &functionImports); err != nil {
+		return err
+	}
+
+	unmarshalFunctionImports := map[string]*functionInfo{}
+	for functionName, functionImport := range functionImports {
+		unmarshalFunctionImports[functionName] = functionImport.Function
+	}
+
+	*fim = unmarshalFunctionImports
+	return nil
+}
+
+type projectImportInfo struct {
+	Project   *projectInfo
+	Functions functionImportMap
```

Comment:
Do we want to import function events from a different system?
To me that sounds a bit weird

Reply:
Sure we do. Why not? function events are presets in the UI to test the function.
This is a really nice convenience tool for the user to invoke it's functions and if the body (which is not structured in nuclio) contains some system specific stuff then, well:
- in scenarios of upgrade we want it
- in scenarios of migration to different system he might have to change some of it, that's ok. it can still contain totally system independent info or a mix of the two in ways we can't predict